lib/Backend/LowerMDSharedSimd128.cpp

```diff
@@ -1327,7 +1336,11 @@ IR::Instr* LowererMD::Simd128LowerShift(IR::Instr *instr)
         IR::RegOpnd * shamtReg = IR::RegOpnd::New(TyInt8, m_func);
         shamtReg->SetReg(LowererMDArch::GetRegShiftCount());
         IR::RegOpnd * tmp = IR::RegOpnd::New(TyInt8, m_func);
-
+#ifdef _M_IX86
```

Comment:
I can take care of pushing this once it's had test coverage. Doug, are watching that?

-- Paul
________________________________
From: Nagy Mostafa<mailto:notifications@github.com>
Sent: ‎3/‎13/‎2016 2:15 PM
To: Microsoft/ChakraCore<mailto:ChakraCore@noreply.github.com>
Cc: Paul Leathers<mailto:pleath@microsoft.com>
Subject: Re: [ChakraCore] SIMD.js - Fixed incorrect operand size in  lowering of shuffle op. Disabled failing shift test on jshost (#549)


In lib/Backend/LowerMDSharedSimd128.cpp<https://github.com/Microsoft/ChakraCore/pull/549#discussion_r55943937>:

> @@ -1327,7 +1336,11 @@ IR::Instr* LowererMD::Simd128LowerShift(IR::Instr *instr)
>          IR::RegOpnd * shamtReg = IR::RegOpnd::New(TyInt8, m_func);
>          shamtReg->SetReg(LowererMDArch::GetRegShiftCount());
>          IR::RegOpnd * tmp = IR::RegOpnd::New(TyInt8, m_func);
> -
> +#ifdef _M_IX86


This done elsewhere in LowererMDShared, and in situations where we need the shift amount to be in cl. So it should work here as well (albeit some potential perf loss, if we have to spill ebx).

I haven't looked into why LinearScan doesn't assign a byteable reg. My guess is that a sym will get byteable reg or not upon its def instr. So if we have Ld d(TyInt8), s(TyInt32), d will get a byteable reg, but the problem is that s will not be moved to a byteable reg, if it is not in one. In other words, I don't think value are moved to a byteable reg if a use demands it.

—
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/ChakraCore/pull/549/files#r55943937>.